### DISCUSSION TITLE
[ROCM] Half bank access conflict.

### DISCUSSION THREAD

**Discussion by LeiWang1999** [ROCm/ROCm#2313] [Q&A] [answered]
Hi all, I'm currently do some micro benchmark of rocm, my device is MI-250, and I've met some issues there, given the program:

```C++
extern "C" __global__ void __launch_bounds__(4) main_kernel0(half* __restrict__ A, half* __restrict__ B) {
  __shared__ half A_shared[16384];
  for (int ax1 = 0; ax1 < 4; ++ax1) {
    for (int ax2_0 = 0; ax2_0 < 64; ++ax2_0) {
      for (int ax3 = 0; ax3 < 64; ++ax3) {
        A_shared[(((ax1 * 4096) + (ax2_0 * 64)) + ax3)] = A[((((((int)blockIdx.x) * 16384) + (ax1 * 4096)) + (ax2_0 * 64)) + ax3)];
      }
    }
  }
  __syncthreads();
  for (int j_0 = 0; j_0 < 64; ++j_0) {
    for (int s = 0; s < 64; ++s) {
      B[((((((int)blockIdx.x) * 16384) + (((int)threadIdx.y) * 4096)) + (j_0 * 64)) + s)] = (A_shared[(((((int)threadIdx.y) * 4096) + (j_0 * 64)) + s)] * __float2half_rn(2.000000e+00f));
    }
  }
}
```

Each threads access 16 bits, while bank controller is 32 bits aligned, which access pattern may cause bank conflict in cuda side, but it seems there's no lds bank conflict when profiled with rocprof, is there any hardware specific design covered this access? Thanks.

**Comment by tcgu-amd** [comment, marked as answer]
Hi @LeiWang1999 Sorry for the lack of response, and thank you for reaching out! Pardon me, but I am a bit confused -- it looks like each thread is accessing consecutive fp16 data in a serialized fashion, since the LSB in each subsequent memory index are determined by the index of the the innermost loop. In this case, I don't think there would be any bank conflict at all -- because no two threads would be trying to access the same bank at the same time. Is my understanding correct?  Once again I would like to apologize of the late response. Thanks!
